uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/Service/WorkPatternCalendar.php

```diff
@@ -314,4 +293,152 @@ private function calculateWorkPatternPeriodEndDate(ContactWorkPattern $contactWo
     return $patternEffectiveEndDate;
   }
 
+  /**
+   * This method fills the dates lapses between the items in the array passed in using
+   * the fabricateMethod.
+   * It also fills lapses between the absence period
+   * start date and the first item in the array and the also between the last item
+   * and the absence period end date using the fabricate method provided.
+   *
+   * The array passed in could be Work Pattern periods
+   * or Contracts.
+   *
+   * @param array $toFabricateFor
+   * @param string $fabricateMethod
+   *
+   * @return array
+   *   An array with the no date lapses in between
+   */
+  private function fabricateForLapses($toFabricateFor, $fabricateMethod) {
+    $fabricated = [];
+
+    if(!method_exists($this, $fabricateMethod)){
+      return $toFabricateFor;
+    }
+    $absencePeriodEndDate = new DateTime($this->absencePeriod->end_date);
+    $absencePeriodStartDate = new DateTime($this->absencePeriod->start_date);
+
+    if(!$toFabricateFor) {
+      $fabricated[] = $this->$fabricateMethod($absencePeriodStartDate, $absencePeriodEndDate);
+      return $fabricated;
+    }
+
+    $firstToFabricateFor = reset($toFabricateFor);
+
+    //fabricate for lapse between beginning of absence period and first $toFabricateFor
+    $firstToFabricateForStartDate = clone $firstToFabricateFor['period_start_date'];
+    if ($firstToFabricateForStartDate > $absencePeriodStartDate) {
+      $fabricated[] = $this->$fabricateMethod($absencePeriodStartDate, $firstToFabricateForStartDate->modify('-1 day'));
+    }
+
+    $toCompare = $firstToFabricateFor;
+    $fabricated[] = $toCompare;
+
+    //fabricate for lapses in between toFabricateFor periods
+    while($next = next($toFabricateFor)) {
+      $intervalInDays = $this->getDateIntervalInDays(
+        $fromDate = clone $toCompare['period_end_date'],
+        $toDate = clone $next['period_start_date']
```

Comment:
Even though I think this is a neat trick, for the sake of readability, I think it would be better to make the assignment outside the function call, to make it more explicit